android/src/main/java/com/tailscale/ipn/ui/localapi/LocalAPIRequest.kt

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2024 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package com.tailscale.ipn.ui.localapi
+
+import android.util.Log
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.decodeFromString
+import com.tailscale.ipn.ui.model.*
+
+class LocalAPIRequest(
+    val path: String,
+    val method: String,
+    val handler: RawAPIResponse,
+) {
+    companion object {
+        var cookieCounter: Int = 0
+
+        fun getCookie(): String {
```

Comment:
Is there potential for data racing here if we have multiple threads creating LocalAPIRequest concurrently?

Reply:
There is!  Fixed.